segmentation/include/pcl/segmentation/impl/extract_labeled_clusters.hpp

```diff
@@ -39,45 +39,99 @@
 
 #include <pcl/segmentation/extract_labeled_clusters.h>
 
+namespace impl
+{
+  // convert labeled_cluster_map_t
+  void
+  labeled_cluster_map_to_labeled_cluster_vector( const pcl::PCLHeader& header, pcl::labeled_cluster_map_t labeled_map, std::vector<std::vector<pcl::PointIndices>>& vec )
+  {
+    if ( labeled_map.empty() )
+      return;
+
+    // get max label
+    const auto extracted_max_label = std::max_element( labeled_map.begin(), labeled_map.end(), 
+      []( const pcl::labeled_cluster_map_t::value_type& lhs, const pcl::labeled_cluster_map_t::value_type& rhs  ) { return lhs.first < rhs.first; }  
+      )
+      ->first;
+
+    // if the user didn't provide a preallocated vector, create it for them based on max label
+    if ( vec.empty() )
+      vec.resize(extracted_max_label+1);
+    else if ( vec.size() <= extracted_max_label )
+    {
+      PCL_ERROR ("[pcl::extractLabeledEuclideanClusters] labeled_clusters size (%lu) less than extracted max_label (%lu)!\n", vec.size (), (unsigned int)extracted_max_label );
+      return;
+    }
+    
+    // move clusters to output vector
+    //  convert pcl::Indices to pcl:PointIndices
```

Comment:
Do you envision this class having 3 extraction methods?  I had assumed 2:  the new `map` method would supercede the `vec-o-vec-o-PointIndices`.  This function exists to provide backcompat.  Or I may be misunderstanding

Reply:
Since you're touching this, I thought it might be easy to add it now, rather than later when someone is going through the interface updating the API.

That means 3 interfaces:
* vec-o-vec-Indices: new
* map-o-vec-Indices: new
* vec-o-vec-PointIndices: old

Reply:
If `vec-o-vec-Indices` will behave similarly to `vec-o-vec-PointIndices` , and therefore inherit the memory inefficiencies in the sparse label case, why not just deprecate and replace with the `map` version?

Actually, in searching for existing usage within PCL, I stumbled across `ConditionalEuclideanClustering`, which seems like a more generic solution to this grouping problem.  Are there any good reasons not to simply deprecate this entire class in favor of  `ConditionalEuclideanClustering` with a trivial label matching function?  Or, under the hood, rewrite this class to use `ConditionalEuclideanClustering`, and then transform the result to comply with existing behavior.   Maybe I'm missing something obvious

Reply:
Wrt deprecation and the alternate of rewriting, I'm not a user of these classes. Hopefully others can chime in